### PR TITLE
[post] action injection - way to legacy or not?

### LINKED ISSUE #299: [post] why action injection in Symfony can turn your application to legacy/laravel
Start with the worst case scenario - inject or autowire everything, find on Nette fórum. Try with controller.action tag

### Resources

- https://www.reddit.com/r/PHP/comments/8dw8x5/symfonys_controller_action_dependency_injection/
- http://paul-m-jones.com/archives/6589
- https://mattstauffer.com/blog/laravel-5.0-method-injection/

### Nette Sources

- https://github.com/geniv/nette-autowired/blob/master/README.md
- https://forum.nette.org/en/1300-nette-framework-2-1-0-finally-released
- https://forum.nette.org/cs/13084-presentery-property-lazy-autowire-na-steroidech#p93574

### PR DESCRIPTION
**Pull request by TomasVotruba**
Closes #299 

## Preview

screenpost

### CODE REVIEW

File: source/_posts/2018/2018-04-23-how-to-slowly-turn-your-symfony-project-to-legacy-with-action-injection.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+```
+
+Can you see the difference to Symfony? Well, almost none. But so far so good.
+
+Note to Nette programmers: [`@inject` is often a code smell and you should do it cleaner](/blog/2016/12/24/how-to-avoid-inject-thanks-to-decorator-feature-in-nette/)
+
+## Inspire by (Good/Bad) Example
+
+If you prepare some "dirty-hack-that-none-should-use" or even better "don't-ever-use-this-unless-you-know-why" and make it public, you can be sure people will ignore it and use it a very creative way. Unless there is `new ForbiddenUseException` thrown.
+
```
Comment (mssimi) [inline source/_posts/2018/2018-04-23-how-to-slowly-turn-your-symfony-project-to-legacy-with-action-injection.md]: Shouldn't it be "use it in a very creative way"?

Comment (TomasVotruba) [reply source/_posts/2018/2018-04-23-how-to-slowly-turn-your-symfony-project-to-legacy-with-action-injection.md]: You're right of course. I'll fix it

How do you understand post as a whole?

Comment (mssimi) [reply source/_posts/2018/2018-04-23-how-to-slowly-turn-your-symfony-project-to-legacy-with-action-injection.md]: Don't overuse action injection?

Comment (TomasVotruba) [reply source/_posts/2018/2018-04-23-how-to-slowly-turn-your-symfony-project-to-legacy-with-action-injection.md]: Better not use it at all

Comment (mssimi) [reply source/_posts/2018/2018-04-23-how-to-slowly-turn-your-symfony-project-to-legacy-with-action-injection.md]: How would you inject Request to controller method? Using constructor RequestStack injection?

Comment (TomasVotruba) [reply source/_posts/2018/2018-04-23-how-to-slowly-turn-your-symfony-project-to-legacy-with-action-injection.md]: Yep. Single approach

**Pull request merged**